app/models/user.rb

```diff
@@ -456,6 +456,16 @@ def close_account!
     AccountDeletion.create(:person => self.person)
   end
 
+  def lock_account!
+    self.locked_at = Time.now
+    self.save
+  end
```

Comment:
Devise already adds this: https://github.com/plataformatec/devise/blob/master/lib/devise/models/lockable.rb#L40